### PR TITLE
Add ticket view

### LINKED ISSUE #1208: Implement ticket view

### PR DESCRIPTION
**Pull request by eatonphil**
Closes #1208 

TODO

* [x] Write tests
* [X] Display attachments -- Note: API doesn't support this yet. Breaking it out into #1501

screen shot 2017-03-31 at 2 05 05 pm
screen shot 2017-03-31 at 2 05 21 pm

### CODE REVIEW

File: src/fetch.js [hunk 1]
```diff
@@ -26,17 +26,29 @@ export function fetch(token, _path, _options) {
...
     ..._options,
     headers: {
       ..._headers,
       Accept: 'application/json',
       Authorization: `token ${token}`,
-      'Content-Type': (_headers && _headers['Content-Type'] || 'application/json'),
     },
   };
 
+  let contentType = 'application/json';
```
Comment (mparke) [inline src/fetch.js]: not a blocker for this PR specifically, 

but thoughts on updating fetch to do a `_.merge` on default and passed options?

simplifies some of the logic, may still need to find an appropriate place for form data defaults

File: src/api/tickets.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { thunkFetchFile } from './apiActionReducerGenerator';
+
+export function addTicketAttachment(id, attachment) {
+  const data = new FormData();
+  data.append('file', attachment);
+  return thunkFetchFile.post(`/support/tickets/${id}/attachments`, data, null);
```
Comment (mparke) [inline src/api/tickets.js]: re: comment on fetch below, would pass options + explicit header change here

File: src/api/apiActionReducerGenerator.js [hunk 3]
```diff
@@ -299,5 +300,16 @@ export const thunkFetch = {
...
   put: _thunkFetch('PUT'),
   get: _thunkFetch('GET'),
-  delete: _thunkFetch('delete'),
+  delete: _thunkFetch('DELETE'),
+};
+
+function _thunkFetchFile(method) {
+  const _fetch = _thunkFetch(method, false);
+  return (url, attachment, type = 'image/png') =>
+    _fetch(url, attachment, { 'Content-Type': type });
```
Comment (mparke) [inline src/api/apiActionReducerGenerator.js]: re: comments on fetch below, would pass form specific options here

File: src/support/components/TicketReply.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  const elements = [];
+  for (let i = 0; i < ps.length; i++) {
+    const p = ps[i];
+
+    if (p === '') {
+      elements.push(<br key={i} />);
+    } else if (p.indexOf('    ') !== -1) {
+      return <pre>{text}</pre>;
```
Comment (mparke) [inline src/support/components/TicketReply.js]: if one case of these spaces is found, we just return the whole chunk in <pre>?

File: src/support/components/TicketReply.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  return (
+    <span>{elements}</span>
+  );
+}
+
+export default function TicketReply(props) {
+  const createdBy = props.reply[`${props.createdField}_by`];
+  const created = moment.utc(props.reply[props.createdField]).fromNow();
```
Comment (mparke) [inline src/support/components/TicketReply.js]: not a blocker for this pr,

is showing the whole date valuable for tickets over a long period/ viewing history of a ticket?

File: src/support/components/TicketReply.js (hunk 4 above)
Comment (eatonphil) [inline src/support/components/TicketReply.js]: Yep. That is the current behavior. I would like to revisit it sometime...

File: src/support/components/TicketReply.js (hunk 5 above)
Comment (eatonphil) [inline src/support/components/TicketReply.js]: My hope for every date is that its shown in relative time by default and hover / tooltip produces an absolute time. Github works like that

File: src/fetch.js (hunk 1 above)
Comment (eatonphil) [inline src/fetch.js]: kk, will address

**Comment by eatonphil** [comment]
thank you!

**Pull request merged**